docs/modules/ROOT/pages/introspection.adoc

```diff
@@ -1,160 +1,68 @@
+Extended Function Selectors, but there are tools that can help with this process, such as {src5-rs}.
 
-[,cairo]
-----
-from openzeppelin.introspection.erc265.IERC165 import IERC165
+=== Registering interfaces
 
-INTERFACE_ID = 0x12345678
+For a contract to declare its support for a given interface, the contract should import the SRC5 module and
+register its support. It's recommended to register interface support upon contract deployment through a constructor
+either directly or indirectly (as an initializer) like this:
 
-func check_support{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    target_contract: felt
-) -> (success: felt) {
-    let (is_supported) = IERC165.supportsInterface(target_contract, INTERFACE_ID);
-    return (is_supported);
-}
+[,javascript]
 ----
+#[starknet::contract]
+mod Contract {
+    use openzeppelin::account::interface;
+    use openzeppelin::introspection::src5::SRC5;
 
-NOTE: `supportsInterface` is camelCased because it is an exposed contract method as part of ERC165's interface.
-This differs from library methods (such as `supports_interface` from the https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/introspection/erc165/library.cairo[ERC165 library]) which are snake_cased and not exposed.
-See the xref:extensibility.adoc#function_names_and_coding_style[Function names and coding style] for more details.
-
-=== IERC165
+    #[storage]
+    struct Storage {}
 
-[,cairo]
-----
-@contract_interface
-namespace IERC165 {
-    func supportsInterface(interfaceId: felt) -> (success: felt) {
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        let mut unsafe_state = SRC5::unsafe_new_contract_state();
+        SRC5::InternalImpl::register_interface(ref unsafe_state, interface::ISRC6_ID);
     }
-}
-----
-
-=== IERC165 API Specification
 
-[,cairo]
-----
-func supportsInterface(interfaceId: felt) -> (success: felt) {
+    (...)
 }
 ----
 
-==== `supportsInterface`
-
-Returns true if this contract implements the interface defined by `interfaceId`.
-
-Parameters:
-
-[,cairo]
-----
-interfaceId: felt
-----
-
-Returns:
-
-[,cairo]
-----
-success: felt
-----
-
-=== ERC165 Library Functions
-
-[,cairo]
-----
-func supports_interface(interface_id: felt) -> (success: felt) {
-}
-
-func register_interface(interface_id: felt) {
-}
-----
-
-[#supportsinterface2]
-==== `supports_interface`
-
-Returns true if this contract implements the interface defined by `interface_id`.
-
-Parameters:
-
-[,cairo]
-----
-interface_id: felt
-----
-
-Returns:
-
-[,cairo]
-----
-success: felt
-----
-
-==== `register_interface`
-
-Calling contract declares support for a specific interface defined by `interface_id`.
-
-Parameters:
+=== Querying interfaces
 
-[,cairo]
-----
-interface_id: felt
-----
+Use the `supports_interface` function to query a contract's support for a given interface.
```

Comment:
This section seems a little bare. In the current OZ Cairo docs, we have an example checking support. Thoughts on including something like that here?

Reply:
That example would be a practical guide (how to), and following the documentation framework should be separated of this "understanding targeted" guide. I don't think we need the practical guide for that though, calling supports_interface should be pretty straightforward after understanding where the ID is coming from.

Reply:
+1 to andrew's suggestion. it does looks bare to me, and right above we're doing the same for `register_interface` which is equally straightforward to understand. i think we need the offer a counterpart.